packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/process.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class ChannelCommand extends FlutterCommand {
+  @override
+  final String name = 'channel';
+
+  @override
+  final String description = 'List or switch flutter channels.';
+
+  @override
+  String get invocation => '${runner.executableName} $name [<channel-name>]';
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  @override
+  Future<int> runInProject() async {
+    switch (argResults.rest.length) {
+      case 0:
+        return await _listChannels();
+      case 1:
+        return await _switchChannel(argResults.rest[0]);
+      default:
+        printStatus('Too many arguments.');
+        printStatus(usage);
+        return 2;
+    }
+  }
+
+  Future<int> _listChannels() async {
+    String currentBranch;
+    await runCommandAndStreamOutput(
+      <String>['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
+      workingDirectory: Cache.flutterRoot,
+      mapFunction: (String line) {
+        currentBranch = line;
+        return null;
+      },
+    );
+
+    printStatus('Flutter channels:');
+    return runCommandAndStreamOutput(
+      <String>['git', 'branch', '-r'],
+      workingDirectory: Cache.flutterRoot,
+      mapFunction: (String line) {
+        List<String> split = line.split('/');
+        if (split.length < 2) return null;
+        String branchName = split[1];
+        if (branchName.startsWith('HEAD')) return null;
+        if (branchName == currentBranch) return '* $branchName';
+        return '  $branchName';
```

Comment:
Good point, but I would not want to hard code a fixed list into flutter. Perhaps there should be a list of prefixes of branches that should be ignored. At the moment, there are only 6 branches in the flutter repo...
```
alpha
master
hackathon
revert-5426-build-sdk-summary
codelab
codelab2
```
I think that `hackathon*` and `codelab*` could be (are?) valid channels that we could use for planned events, but `revert*` should definitely be excluded.

We could ignore any branches that start with
* revert
* test
* any others?

@sethladd Any thoughts on what branches should be listed / excluded?